pkg/fleetautoscalers/controller.go

```diff
@@ -292,11 +282,6 @@ func (c *Controller) syncFleetAutoscaler(ctx context.Context, key string) error
 	}
 
 	if fas == nil {
-		// just in case we don't catch a delete event for some reason, use this as a
-		// failsafe to ensure we don't end up leaking goroutines.
-		if runtime.FeatureEnabled(runtime.FeatureCustomFasSyncInterval) {
-			return c.cleanFasThreads(key)
```

Comment:
`return c.cleanFasThreads(key) ` should have stayed in this block.

Reply:
I've updated with this correct statement